### PR TITLE
Docs: Keycloak OIDC Guide

### PR DESCRIPTION
**Pull request by mirceanton**
Addresses #4109

**What changed?**

Improved documentation to contain a step-by-step guide of configuring the Keycloak OIDC integration.

**Why was this change made?**

The documentation for the generic OIDC setup was not clear enough as to make the process seamless.

**How did you validate the change?**

By following the steps described in the guide 😆 

**Release notes**

**Documentation Changes**

Not really, it's just docs.

### CODE REVIEW

**Comment by mirceanton** [comment]
@bigkevmcd here is the PR for the Keycloak guide we talked about!

**Comment by mirceanton** [comment]
I need to look a bit more into the file structure and the actual website config as I am not familiar with docusaurus at all 😅 
But the content itself is pretty much done

**Review by bigkevmcd — APPROVED** [review]
This looks fantastic, I'll find out "where" to put it, but otherwise I'd be happy to accept this, thanks for the work.

File: website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+      auth:
+        username: bn_keycloak
+        database: bitnami_keycloak
+        password: potgres123  # <-- make use of `existingSecret` functionality to hide passwords
+        postgresPassword: postgres123  # <-- make use of `existingSecret` functionality to hide passwords
+```
+
+</details>
+
+### Configuring a new Keycloak Realm
```
Comment (makkes) [inline website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx:112]:
```suggestion
## Configuring a new Keycloak Realm
```

IMHO this guide should focus on how to configure keycloak for the specific use case of using it as an IdP.

File: website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx [hunk 2]
```diff
@@ -0,0 +1,367 @@
+---
+title: Configuring OIDC with Keycloak
+---
+
+In this guide we will show you how to enable users to login to the Weave GitOps dashboard by authenticating them against a Keycloak instance.
+
+This example uses [Keycloak](https://www.keycloak.org/) and assumes Weave GitOps has already been installed on the Kubernetes Cluster.
+
+### Pre-requisites
```
Comment (makkes) [inline website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx:9]:
```suggestion
## Pre-requisites
```

File: website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+- A Kubernetes cluster runnnig a [Flux-supported version of Kubernetes](https://fluxcd.io/docs/installation/#prerequisites)
+- Weave GitOps is [installed](https://docs.gitops.weave.works/docs/installation/) and [TLS has been enabled](https://docs.gitops.weave.works/docs/configuration/tls/)
```
Comment (makkes) [inline website/docs/guides/sso/keycloak/configuring-oidc-with-keycloak.mdx:12]:
```suggestion
- Weave GitOps is [installed](https://docs.gitops.weave.works/docs/installation/) and [TLS has been enabled](https://docs.gitops.weave.works/docs/configuration/tls/)
- Access to a running Keycloak installation
```

**Review by makkes — COMMENTED** [review]
Thanks for this wonderful guide. I left some comments.

@bigkevmcd as to where to leave this, I think after merging in #3917 we can incorporate it either into that page or as a distinct page and link to it from the general OIDC guide.

**Comment by mirceanton** [comment]
@makkes I will look on the comments this weekend and fix everything up! Thanks

**Comment by makkes** [comment]
@mirceanton can you please rebase your branch on top of `main`, get rid of the merge commits and squash the commits into a single one?

**Comment by mirceanton** [comment]
@makkes done. Rebased and squashed. Though I must ask, why squash the commits on the MR instead of `git merge --squash`?

**Comment by makkes** [comment]
I replaced the Keycloak guide in the OIDC guide with a link to this new page and also moved this page to the top-level section for simplicity. Great job @mirceanton! /cc @bigkevmcd

File: website/docs/guides/configuring-oidc-with-keycloak.mdx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+After this secret is created, you may need to delete the Weave GitOps pods in order to restart the app and load the new config.
+
+### Setting up RBAC
+
+Once Weave GitOps is configured for OIDC, we need a way to map permissions to the groups. To do that, we need to create role bindings for our `wego-admin` group.
+The following example assumes that the ClusterRole `wego-admin-cluster-role` and the namespaced Role `wego-admin-role` already exist. See the
+[recommendations on setting up RBAC](../../enterprise/getting-started/install-enterprise/#recommended-rbac-configuration) for details.
+
+<details><summary>Expand to see group role bindings</summary>
```
Comment (makkes) [inline website/docs/guides/configuring-oidc-with-keycloak.mdx:223]: I'm not sure I understand why in this example the user would need to create a namespaced and a cluster-scoped binding. The ClusterRoleBinding should be sufficient in this case.

Comment (mirceanton) [reply website/docs/guides/configuring-oidc-with-keycloak.mdx:223]: I believe it was a bad copy-paste job from here: https://github.com/weaveworks/weave-gitops/blob/main/website/versioned_docs/version-0.36.0/enterprise/getting-started/install-enterprise.mdx#L841-L869
Or rather, the configuration I ended up with after tinkering with that example for a while and I forgot to proofread it 😅

**Review by makkes — APPROVED** [review]
Ready to merge.

**Comment by makkes** [comment]
Thanks a lot, @mirceanton. This will definitely save users from a lot of headaches. 👏🏻 🎖️

**Pull request merged**